lib/main.js

```diff
@@ -112,67 +166,76 @@ export default {
     tslintDef = require('loophole').allowUnsafeNewFunction(() => require(TSLINT_MODULE_NAME).Linter);
 
     return {
+      name: 'TSLint',
       grammarScopes: ['source.ts', 'source.tsx'],
       scope: 'file',
       lintOnFly: true,
       lint: (textEditor) => {
-        const filePath = textEditor.getPath();
-        const text = textEditor.getText();
-
-        return this.getLinter(filePath).then((Linter) => {
-          const configurationPath = Linter.findConfigurationPath(null, filePath);
-          const configuration = Linter.loadConfigurationFromPath(configurationPath);
-
-          let { rulesDirectory } = configuration;
-          if (rulesDirectory) {
-            const configurationDir = path.dirname(configurationPath);
-            if (!Array.isArray(rulesDirectory)) {
-              rulesDirectory = [rulesDirectory];
-            }
-            rulesDirectory = rulesDirectory.map((dir) => {
-              if (path.isAbsolute(dir)) {
-                return dir;
-              }
-              return path.join(configurationDir, dir);
-            });
-
-            if (this.rulesDirectory) {
-              rulesDirectory.push(this.rulesDirectory);
-            }
-          }
+        if (atom.config.get('linter-tslint.ignoreTypings') && /.*\.d\.ts$/i.test(textEditor.getPath())) {
```

Comment:
Also, how about this for the second part? If the case-insensitivity is important add a `.toLowerCase()` in there.

```js
textEditor.getPath().endsWith('.d.ts')
```